utils/mgmt_system/kubernetes.py

```diff
@@ -0,0 +1,181 @@
+from collections import namedtuple
+from mgmtsystem.base import ContainerMgmtSystemAPIBase
+from mgmtsystem.rest_client import ContainerClient
+
+"""
+Related yaml structures:
+
+[cfme_data]
+management_systems:
+    kubernetes:
+        name: My kubernetes
+        type: kubernetes
+        hostname: 10.12.13.14
+        port: 6443
+        credentials: kubernetes
+        authenticate: true
+        rest_protocol: https
+
+[credentials]
+kubernetes:
+    username: admin
+    password: secret
+    token: mytoken
+"""
+
+# Below are simplified representations of CFME k8s objects that should cover most of our needs
+Container = namedtuple('Container', ['name', 'cg_name', 'image'])
+Image = namedtuple('Image', ['name', 'tag', 'id'])
+# Images are equal if their ids match
+Image.__eq__ = lambda self, other: self.id == other.id
+ImageRegistry = namedtuple('ImageRegistry', ['host', 'port'])
+Project = namedtuple('Project', ['name'])
+Node = namedtuple('Node', ['name', 'ready', 'cpu', 'memory'])
+ContainerGroup = namedtuple(
+    'ContainerGroup', ['name', 'project_name', 'restart_policy', 'dns_policy'])
+ReplicationController = namedtuple(
+    'ReplicationController', ['name', 'project_name', 'replicas', 'current_replicas'])
+Service = namedtuple('Service', ['name', 'project_name', 'portal_ip', 'session_affinity'])
+
+
+class Kubernetes(ContainerMgmtSystemAPIBase):
+
+    _stats_available = {
+        'num_container': lambda self: len(self.list_container()),
+        'num_pod': lambda self: len(self.list_container_group()),
+        'num_service': lambda self: len(self.list_service()),
+        'num_replication_controller':
+            lambda self: len(self.list_replication_controller()),
+        'num_image': lambda self: len(self.list_image()),
+        'num_node': lambda self: len(self.list_node()),
+        'num_image_registry': lambda self: len(self.list_image_registry()),
+        'num_project': lambda self: len(self.list_project()),
+    }
+
+    def __init__(self, hostname, protocol="https", port=6443, entry='api/v1', **kwargs):
+        self.hostname = hostname
+        self.username = kwargs.get('username', '')
+        self.password = kwargs.get('password', '')
+        self.token = kwargs.get('token', '')
+        self.auth = self.token if self.token else (self.username, self.password)
+        self.api = ContainerClient(hostname, self.auth, protocol, port, entry)
+
+    def disconnect(self):
+        pass
+
+    def _parse_image_info(self, image_str):
+        """Splits full image name into registry, name and tag
+
+        Both registry and tag are optional, name is always present.
+
+        Example:
+            localhost:5000/nginx:latest => localhost:5000, nginx, latest
+        """
+        registry, image_str = image_str.split('/', 1) if '/' in image_str else ('', image_str)
+        name, tag = image_str.split(':') if ':' in image_str else (image_str, '')
```

Comment:
I think it might do `.split(':', 1)` as the split in the previous line to avoid the tuple expansion error.